src/LargeFileUploadTask.ts

```diff
@@ -0,0 +1,338 @@
+                let nextRange = self.getNextRange();
+                if (nextRange.maxValue === -1) {
+                    let err = new Error("Task with which you are trying to upload is already completed, Please check for your uploaded file");
+                    err.name = "Invalid Session";
+                    throw err;
+                }
+                let fileSlice = self.sliceFile(nextRange),
+                    response = await self.uploadSlice(fileSlice, nextRange, self.file.size);
+                // Upon completion of upload process incaseof onedrive, driveItem is returned, which contains id
+                if (response.id !== undefined) {
+                    return response;
+                } else {
+                    self.updateTaskStatus(response);
+                }
+            }
+        } catch(err) {
+            throw err;
+        }
+    }
+
+    /**
+     * @async
+     * Upload the particular slice to the server
+     * @param fileSlice - The file slice
+     * @param range - The range value
+     * @param totalSize - The size of the complete file
+     */
+    async uploadSlice(fileSlice: ArrayBuffer | Blob | File, range: Range, totalSize: number): Promise<any> {
+        let self = this;
+        try {
+            if (self.uploadSession.expiry.getTime() <= Date.now()) {
+                let err = new Error("Task with which you are uploading is no longer valid, Please create new task to upload");
+                err.name = "Invalid Session";
+                throw err;
+            }
+            return await self.client
+                .api(self.uploadSession.url)
+                .headers({
+                    "Content-Length": `${range.maxValue - range.minValue + 1}`,
+                    "Content-Range": `bytes ${range.minValue}-${range.maxValue}/${totalSize}`
+                })
+                .put(fileSlice);
+        } catch(err) {
+            throw err;
+        }
+    }
+
+    /**
+     * @async
+     * Deletes the upload session in the server
+     * @return The promise resolves to cancelled response
+     */
+    async cancel (): Promise<any> {
+        let self = this;
+        try {
+            return await self.client
+                .api(self.uploadSession.url)
+                .delete();
+        } catch (err) {
+            throw err;
+        }
+    }
+
+    /**
+     * @async
+     * Get the status for the upload session
+     * @return The promise resolves to the status enquiry response
+     */
+    async getStatus(): Promise<any> {
+        let self = this;
+        try {
+            let response = await self.client
+                            .api(self.uploadSession.url)
+                            .get();
+            self.updateTaskStatus(response);
+            return response;
+        } catch (err) {
+            throw err;
+        }
+    }
+
+    /**
+     * @async
+     * Resume the upload session and continue uploading the file from the place where it left
```

Comment:
Resume the upload session and continue uploading the file from the last sent range.